hydra-node/test/Hydra/Chain/DirectSpec.hs

```diff
@@ -3,58 +3,68 @@
-import Hydra.Logging (nullTracer)
+import Hydra.Ledger.Cardano (CardanoTx, NetworkMagic)
+import Hydra.Logging (showLogsOnFailure)
 import Hydra.Party (Party, deriveParty, generateKey)
 import Test.QuickCheck (generate)
 
 spec :: Spec
 spec = do
-  it "can init and abort a head given nothing has been committed" $ do
-    calledBackAlice <- newEmptyMVar
-    calledBackBob <- newEmptyMVar
-    aliceKeys@(aliceVk, _) <- generateKeyPair
-    bobKeys@(bobVk, _) <- generateKeyPair
-    withMockServer $ \magic iocp socket submitTx -> do
-      let cardanoKeys = [] -- TODO(SN): this should matter
-      withDirectChain nullTracer magic iocp socket aliceKeys alice cardanoKeys (putMVar calledBackAlice) $ \Chain{postTx} -> do
-        withDirectChain nullTracer magic iocp socket bobKeys bob cardanoKeys (putMVar calledBackBob) $ \_ -> do
-          let parameters = HeadParameters 100 [alice, bob, carol]
-          generate (genPaymentTo magic aliceVk) >>= submitTx
-          generate (genPaymentTo magic bobVk) >>= submitTx
-
-          failAfter 5 $
-            retrying @ErrorCall $ postTx $ InitTx parameters
-          failAfter 5 $
-            takeMVar calledBackAlice `shouldReturn` OnInitTx 100 [alice, bob, carol]
-          failAfter 5 $
-            takeMVar calledBackBob `shouldReturn` OnInitTx 100 [alice, bob, carol]
-
-          postTx $ AbortTx mempty
-
-          failAfter 5 $
-            takeMVar calledBackAlice `shouldReturn` OnAbortTx
-          failAfter 5 $
-            takeMVar calledBackBob `shouldReturn` OnAbortTx
+  it "can init and abort a head given nothing has been committed" $
+    failAfter 10 $
+      showLogsOnFailure $ \tracer -> do
+        calledBackAlice <- newEmptyMVar
+        calledBackBob <- newEmptyMVar
+        aliceKeys@(aliceVk, _) <- generateKeyPair
+        bobKeys <- generateKeyPair
+        withMockServer $ \magic iocp socket submitTx -> do
+          let cardanoKeys = [] -- TODO(SN): this should matter
+          withDirectChain (contramap FromAlice tracer) magic iocp socket aliceKeys alice cardanoKeys (putMVar calledBackAlice) $ \Chain{postTx} -> do
+            withDirectChain (contramap FromBob tracer) magic iocp socket bobKeys bob cardanoKeys (putMVar calledBackBob) $ \_ -> do
+              let parameters = HeadParameters 100 [alice, bob, carol]
+              mkSeedPayment magic aliceVk submitTx
+
+              retry (== NoSeedInput @CardanoTx) $ postTx $ InitTx parameters
+              takeMVar calledBackAlice `shouldReturn` OnInitTx 100 [alice, bob, carol]
+              takeMVar calledBackBob `shouldReturn` OnInitTx 100 [alice, bob, carol]
+
+              postTx $ AbortTx mempty
+              takeMVar calledBackAlice `shouldReturn` OnAbortTx
+              takeMVar calledBackBob `shouldReturn` OnAbortTx
+
+mkSeedPayment :: NetworkMagic -> VerificationKey -> (ValidatedTx Era -> IO ()) -> IO ()
+mkSeedPayment magic vk submitTx =
+  generate (genPaymentTo magic vk) >>= submitTx
```

Comment:
This name conflicts with the other, real `mkSeedPayment` .. this confused me once already.